File: cucumber/ruby/calculator/features/step_definitions/calculator_steps.rb
require './calc.rb';

Given /^the input "([^"]*)"$/ do |arg1|
    @input = arg1;
end

When /^the calculator is run$/ do
    @output = Calc.compute(@input);
end

Then /^the output should be "([^"]*)"$/ do |arg1|
    @output.to_i.should == arg1.to_i;
end

File: cucumber/perl/calculator/features/step_definitions/calculator_steps.pl
#!/bin/env perl
use strict;

use Calc;

use Test::More;
use Test::BDD::Cucumber::StepFile;
use Method::Signatures; # Allows func($c) {} syntax
################################################################################
Given qr/^the input "([^"]*)"$/, func ($c) {
    $c->stash->{scenario}->{input} = $1;
};

When qr/^the calculator is run$/, func ($c) {
    my $input = $c->stash->{scenario}->{input};
    $c->stash->{scenario}->{output} = Calc->compute($input);
};

Then qr/^the output should be "([^"]*)"$/, func($c) {
    my $output = $c->stash->{scenario}->{output};
    is $output, $1;
};
################################################################################
